File: js/src/service/UserService.js
const UserRepository = require('../repository/UserRepository');

class UserService {

  constructor(UserRepository){
    this.userRepository = UserRepository;
  }

  create(user) { 
    const age = user.age;
  
    if (age < 18) {
        throw new ApplicationException("Cadastro inválido, usuário tem menos de 18 anos");
    
    } else if (age > 75) {
        throw new ApplicationException("Cadastro inválido, usuário tem mais de 75 anos");
  
    }
  
    return this.userRepository.create(user);
  }
  
  getAll() {
    return this.userRepository.getAll();
  }
  
  getById(id) { 
    return this.userRepository.getById(id);
  }
  
  update(user, id) { 
    return this.userRepository.update(user, id);
  }
  
  deleteById(id) { 
    if (this.userRepository.existsById(id)) {
      this.userRepository.deleteById(id);

    } else {
      throw new NotFoundException("Id do recurso não existente");
    }
  }
}
  
function ApplicationException(message) {
  this.message = message;
  this.name = "ApplicationException";
  this.code = 400;
}
  
function NotFoundException(message) {
  this.message = message;
  this.name = "NotFoundException";
  this.code = 404;
}

module.exports = UserService;

File: csharp/Mock/Services/UserService.cs
using System;
using System.Linq;
using System.Collections.Generic;
using Mock.Models;
using Mock.Repositories;

namespace Mock.Services
{
    public class UserService : IUserService
    {
        private readonly IUserRepository _userRepository;

        public UserService(IUserRepository userRepository)
        {
            _userRepository = userRepository;
        }

        public IEnumerable<User> GetUsers()
        {
            return _userRepository.GetUsers();
        }

        public User GetUser(long id)
        {
            return _userRepository.GetUser(id);
        }

        public User CreateUser(User user)
        {
            int age = user.Age;

            if (age < 18) {
                throw new ApplicationException("Cadastro inválido, usuário tem menos de 18 anos");
            
            } else if (age > 75) {
                throw new ApplicationException("Cadastro inválido, usuário tem mais de 75 anos");

            }

            return _userRepository.CreateUser(user);
        }

        public User UpdateUser(User user, long id)
        {
            return _userRepository.UpdateUser(user, id);
        }

        public void DeleteUser(long id)
        {
            _userRepository.DeleteUser(id);
        }
    }
}